### PR TITLE
Added some helpers for string manipulation.

### PR DESCRIPTION
**Pull request by SimonPStevens**
**Parent issue** 

Closes https://github.com/mockoon/mockoon/issues/413

**Technical implementation details**

Added some string manipulation helpers that just call the equivalent Javascript methods directly.

### CODE REVIEW

**Comment by 255kb** [comment]
Thank you @SimonPStevens for these helpers :) 
Like for the `dateTimeShift` helper, you should account for the case where no params are passed. We currently have an error with the `indexOf` (didn't test the other helpers) when used like this: `{{indexOf}}`. 
Could you please also add more test cases? Like for the `setVar`: one case without param, with only the first param, with string param, number param, another embedded helper (`indexOf (body 'prop') 'a'`), etc. Helpers code is a little bit easy to break and I want it to be covered as much as possible. 
Thanks!

**Comment by SimonPStevens** [comment]
Yeah, no prob. I'll get to this later in the week.

**Comment by SimonPStevens** [comment]
Hey. Sorry it took me way longer than I said to get round to doing this. I've added some more unit tests, and fixed the error when the wrong parameter type is passed. Let me know if there is anything else that needs doing.

**Comment by SimonPStevens** [comment]
Actually. Hold off. This still doesn't work with variables, and I'm not quite sure why. I'm still working on it.

**Comment by SimonPStevens** [comment]
OK. I think this is good now.

That opened up a bit of a can of worms with the different parameter types that can be passed to the helpers. I've covered everything I can think of now, but if you can see anything I've missed then let me know.

File: src/libs/template-parser.ts [hunk 1]
```diff
@@ -335,6 +335,83 @@ const TemplateParserHelpers = function (request: Request) {
...
+          : search.toString();
+
+      position =
+        (typeof position === 'object' || typeof position === 'undefined') &&
+        !(position instanceof SafeString)
+          ? undefined
+          : Number(position.toString());
+
+      if (typeof position === 'number') {
+        return data.toString().indexOf(search.toString(), position);
```
Comment (255kb) [inline src/libs/template-parser.ts:362]: `data` and `search` are already converted to string above
```suggestion
        return data.indexOf(search, position);
```

File: src/libs/template-parser.ts [hunk 2]
```diff
@@ -335,6 +335,83 @@ const TemplateParserHelpers = function (request: Request) {
...
+      } else {
+        return data.toString().indexOf(search.toString());
```
Comment (255kb) [inline src/libs/template-parser.ts:364]: `data` and `search` are already converted to string above
```suggestion
        return data.indexOf(search);
```

File: src/libs/template-parser.ts [hunk 3]
```diff
@@ -335,6 +335,83 @@ const TemplateParserHelpers = function (request: Request) {
...
+          ? ''
+          : data.toString();
+
+      search =
+        (typeof search === 'object' || typeof search == 'undefined') &&
+        !(search instanceof SafeString)
+          ? ''
+          : search.toString();
+
+      return data.toString().includes(search.toString());
```
Comment (255kb) [inline src/libs/template-parser.ts:384]: `data` and `search` are already converted to string above
```suggestion
      return data.includes(search);
```

**Review by 255kb — CHANGES_REQUESTED** [review]

**Comment by 255kb** [comment]
@SimonPStevens It looks good. Just 3 `toString()` that seems redundant to me. Then it's ready to be merged!

**Comment by SimonPStevens** [comment]
@255kb All done.

**Comment by 255kb** [comment]
Thanks @SimonPStevens !

**Review by 255kb — APPROVED** [review]

**Pull request merged**